onnxscript/irbuilder.py

```diff
@@ -160,7 +160,7 @@ def to_node_proto(self, node_name: str) -> onnx.NodeProto:
         n = helper.make_node(
             self.callee.opname,
             [_opt_var_to_str(x) for x in self.args],
-            [str(x) for x in self.result],
+            list(self.result),
```

Comment:
Why is this correct? It seems to rely on the type of `self.result` and may fail if that is changed from string to something else (like an IRVar), right?

Reply:
I think self.result was annotated as Sequence[str], should we update the annotation?

Reply:
My point was that this change is not robust. I don't think there is a good reason to make this change even if the annotation is correct today.

Reply:
OK. I will change this back

Reply:
Looking at this again, I still think we should maintain the invariant at the setter/initializer if we annotate a field as a certain type. This way users can assume the invariant and not have to worry about the actual type being different than what's claimed.

Reply:
Okay. Feel free to update the annotation to indicate that it is string-like: something that can be converted to a string via "str(.)".

But if that annotation is "Any", that is probably not a great annotation.

I think the normal subtyping safety principle is: callers should absolutely respect the annotation on parameters, but if the called code handles a superset of the annotated-type, that is perfectly fine and safe.

Reply:
I recall all Python objects have __str__ defined so that basically equates to Any

Reply:
Also: I believe that it is unsafe for `refurb` to do such a transformation, in general, because python does not *require* that type-annotations be satisfied, so the transformation can change the behavior of python-code, which is generally not a good choice for such tools. Any "cleanup" tool should play it safe and avoid transformations that might change the behavior.

Reply:
I think it makes sense to annotate as any if that’s the most accurate type. That said I will revert this change for now!

Reply:
I did the change because I saw the annotation. Refurb didn’t do it :D

Reply:
done